### DISCUSSION TITLE
[BUG] Pusher Swift SDK can't connect to Soketi Serverless

### DISCUSSION THREAD

**Discussion by buttjer** [soketi/pws#929] [Cloudflare Workers: Support]
**Description**
Can't connect. Stuck in connecting state and throwing this error. 
``` 
nw_protocol_copy_ws_definition_block_invoke [C1.1.1:1] nw_ws_validate_server_response
``` 

**Reproduction steps**
Install latest Pusher Swift SDK.

```swift
        let builder = AuthRequestBuilder(authUrl: URL(string: "https://localhost/broadcasting/auth")!)

        let options = PusherClientOptions(
            authMethod: .authRequestBuilder(authRequestBuilder: builder),
            attemptToReturnJSONObject: true,
            autoReconnect: true,
            host: PusherHost.host("soketi-***.***.workers.dev"),
            port: 443,
            useTLS: true
        )

        pusher = Pusher(key: "***", options: options)
       pusher.connect()
``` 

leads to

``` 
nw_protocol_copy_ws_definition_block_invoke [C1.1.1:1] nw_ws_validate_server_response
``` 

and with logging

```swift
    public func changedConnectionState(from old: ConnectionState, to new: ConnectionState) {
        log.debug("Changed connection state from \(connectionStateString(forConnectionState: old)) to \(connectionStateString(forConnectionState: new))")
    }
``` 

only state change to .connecting was logged, never .connected

``` 
Changed connection state from Disconnected to Connecting
``` 

**Expected behavior**
connects

**Environment**
- Soketi Serverless 0.7.0
- Pusher Channels Swift Client 10.1.3 (https://github.com/pusher/pusher-websocket-swift)

**Related Issues**
https://github.com/pusher/pusher-websocket-swift/issues/372 
https://github.com/pusher/pusher-websocket-swift/issues/352
https://github.com/chinloyal/pusher_client/issues/36
https://github.com/pusher/pusher-websocket-swift/issues/328
https://github.com/soketi/soketi/issues/585

**Additional context**
You could assume that Pusher altered their Swift SDK in a way its now incompatible with third party hosts with version 9+.
Or Soketi is answering not as expected in an early connecting state.

**Comment by rennokki** [comment]
Hi there, the issues are not meant for Cloudflare Serverless versions. I will gladly triage this and look on it further soon.

**Comment by alexmck** [comment]
@buttjer Since MessageBird purchased Pusher, this seems to be a deliberate change in direction for Pusher, where they no longer support customer servers in newer versions of their Swift SDK. If you want to use the Pusher Swift SDK with Soketi, you will need to use version 8 in your project.

**Comment by buttjer** [comment]
@alexmck I don't think so. They changed the underlying connection(socket?) package and it's might just not compatible anymore.

> Thanks. We replaced the underlying package that initiates the connection request in v9 so this will be the cause of the issue - however as this works well with our own service our advice would be to continue to use v8 as we can't commit to supporting custom websocket server connections. – @benw-pusher
